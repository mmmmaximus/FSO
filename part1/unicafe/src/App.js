import React, { useState } from 'react'
import Header from './components/Header'
import Button from './components/Button'
import Statistics from './components/Statistics'

const App = () => {
  const [good, setGood] = useState(0)
  const [neutral, setNeutral] = useState(0)
  const [bad, setBad] = useState(0)

  return (
    <div data-testid='App'>
      <Header header='give feedback'/>
      <Button text='good' onClick={() => setGood(good +1)}/>
      <Button text='neutral' onClick={() => setNeutral(neutral +1)}/>
      <Button text='bad' onClick={() => setBad(bad +1)}/>
      <Header header='statistics'/>
      <Statistics good={good} neutral={neutral} bad={bad}/>
    </div>
  )
}

export default App
