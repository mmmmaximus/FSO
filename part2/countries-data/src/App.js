import React, {useState, useEffect} from 'react';
import axios from 'axios';
import Filter from './components/Filter'
import Countries from './components/Countries'

const App = () => {
  const [filter, setFilter] = useState('')
  const [countries, setCountries] = useState([])
  
  const changeFilter = (event) => {
    setFilter(event.target.value)
  }

  useEffect(() => {
    axios
      .get('https://restcountries.eu/rest/v2/all')
      .then(response => {
        setCountries(response.data)
      })
  }, [])
  
  return (
    <div>
      <Filter func={changeFilter}/>
      <Countries 
        countries={
          countries.filter(
            country => country.name.toLowerCase().includes(filter.toLowerCase())
          )
        }
        filter={filter}
      />
    </div>
  )
}

export default App;
