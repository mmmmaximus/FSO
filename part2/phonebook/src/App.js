import React, { useState, useEffect } from 'react'
import Input from './components/Input'
import PersonForm from './components/PersonForm'
import Persons from './components/Persons'
import ErrorMessage from './components/ErrorMessage'
import phonebookService from './services/phonebook'

const App = () => {
  const [persons, setPersons] = useState([])
  const [newFilter, setNewFilter] = useState('')
  const [newName, setNewName] = useState('')
  const [newNumber, setNewNumber] = useState('')
  const [errorMessage, setErrorMessage] = useState('')
  const [messageType, setMessageType] = useState('')
  const changeFilter = (event) => {
    setNewFilter(event.target.value)
  }
  const changeNumber = (event) => {
    setNewNumber(event.target.value)
  }
  const changeName = (event) => {
    setNewName(event.target.value)
  }
  const clearInputs = () => {
    setNewName('')
    setNewNumber('')
    document.getElementById('name').value = ''
    document.getElementById('number').value = ''
  }

  useEffect(() => {
    phonebookService
      .index()
      .then(persons => setPersons(persons))
  }, [])

  const addContact = (event) => {
    event.preventDefault()
    if (newName === '' || newNumber === '') {
      alert(`Name and Number must be filled in`)
    } else if (persons.map(person => person.name.toLowerCase()).includes(newName.toLowerCase())) {
      if (persons.map(person => person.number).includes(newNumber)) {
        alert(`${newName} is already added to phonebook`)
      } else {
        if (window.confirm(`${newName} is already added to phonebook, replace old number?`)) {
          const changedPerson = persons.find(person => person.name === newName)
          phonebookService
            .update(changedPerson.id, { name: newName, number: newNumber })
            .then(person => {
              setPersons(persons.filter(person => person.name !== newName).concat(person))
              setMessageType('notification')
              setErrorMessage(`${person.name} has been updated`)
              setTimeout(() => {
                setErrorMessage('')
              }, 1000)
            }).catch(err => {
              setMessageType('error')
              setErrorMessage(`${newName} has already been deleted`)
              setTimeout(() => {
                setErrorMessage('')
              }, 1000)
            })
          clearInputs()
        }
      }
    } else {
      phonebookService
        .create({ name: newName, number: newNumber })
        .then(person => {
          setPersons(persons.concat(person))
          setMessageType('notification')
          setErrorMessage(`${person.name} has been added`)
          setTimeout(() => {
            setErrorMessage('')
          }, 1000)
        })
      clearInputs()
    }
  }

  const deletePerson = (id, name) => {
    if (window.confirm(`Delete ${name}?`)) {
      phonebookService
        .del(id)
        .then(phonebookService
          .index()
          .then(persons => {
            setPersons(persons.filter(person => person.id !== id))
            setMessageType('notification')
            setErrorMessage(`${name} has been deleted`)
            setTimeout(() => {
              setErrorMessage('')
            }, 1000)
          })
        )
    }
  }

  return (
    <div>
      <h2>Phonebook</h2>
      <ErrorMessage message={errorMessage} type={messageType}/>
      <Input text='filter shown with ' func={changeFilter} />
      <h2>add a new</h2>
      <PersonForm changeName={changeName} changeNumber={changeNumber} addContact={addContact}/>
      <h2>Numbers</h2>
      <Persons deletePerson={deletePerson} persons={
        persons.filter(
          person => person.name.toLowerCase().includes(newFilter.toLowerCase())
        )
      }/>
      ...
    </div>
  )
}

export default App
