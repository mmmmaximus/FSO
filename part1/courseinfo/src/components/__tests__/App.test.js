import { render, screen } from '@testing-library/react'
import { within } from '@testing-library/dom'
import App from './../../App'
import renderer from 'react-test-renderer'

it('should render App component', () => {
  const course = {
    name: 'Half Stack application development',
    parts: [
      {
        name: 'Fundamentals of React',
        exercises: 10
      },
      {
        name: 'Using props to pass data',
        exercises: 7
      },
      {
        name: 'State of a component',
        exercises: 14
      }
    ]
  }
  render(<App />)
  const AppElement = screen.getByTestId('App')
  expect(AppElement).toBeInTheDocument

  // render contents of App component
  const HeaderInContent = within(AppElement).getAllByTestId('Header')
  expect(HeaderInContent.length).toBe(1)
  const ContentInContent = within(AppElement).getAllByTestId('Content')
  expect(ContentInContent.length).toBe(1)
  const TotalInContent = within(AppElement).getAllByTestId('Total')
  expect(TotalInContent.length).toBe(1)
})

it('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
