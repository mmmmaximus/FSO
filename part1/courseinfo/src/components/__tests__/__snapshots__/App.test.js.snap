// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="App"
>
  <h1
    data-testid="Header"
  >
    Half Stack application development
  </h1>
  <div
    data-testid="Content"
  >
    <p
      data-testid="Part"
    >
      Fundamentals of React
       
      10
    </p>
    <p
      data-testid="Part"
    >
      Using props to pass data
       
      7
    </p>
    <p
      data-testid="Part"
    >
      State of a component
       
      14
    </p>
  </div>
  <p
    data-testid="Total"
  >
    Number of exercises 
    31
  </p>
</div>
`;
